### PR TITLE
Game add flag view for route Destination

### PR DESCRIPTION
**Pull request by Noriakisana**
- [ ] TODO: Sort the destinations based on colours and numbers
- [x] TOOD: formatting better

### CODE REVIEW

**Review by eyip002 — CHANGES_REQUESTED** [review]
Instead of generating an SVG for every possible destination symbol, which every device must download, why not have one SVG that can be styled with the appropriate colours and number circles? 
1. Generate an SVG template for a dice with 5 circles (give each circle an ID)
2. Create CSS to define the possible colours and to show or hide each circle
3. Generate the signal-symbol JSON file, where each signal is associated with the required CSS styles

See here for information on styling SVG elements with CSS: 
- [See "Inline stylesheets with the style tag"](https://www.educative.io/answers/how-to-manipulate-svgs-via-css)
- [Modify SVG element classes with JavaScript](https://medium.com/alistapart/javascript-modify-svg-element-class-4ca3055a2024)

File: server/src/assets/game/script-game.js [hunk 1]
```diff
@@ -40,20 +44,35 @@ function disableAllDestinationButtons() {
 		$(`#${destinationNamePrefix}${i}`).prop('disabled', true);
 		$(`#${destinationNamePrefix}${i}`).removeClass(destinationEnabledButtonStyle);
 		$(`#${destinationNamePrefix}${i}`).addClass(destinationDisabledButtonStyle);
+		document.getElementById(`${destinationNamePrefix}${i}`).innerHTML = "<p>Keine Route</p>";
```
Comment (eyip002) [inline server/src/assets/game/script-game.js:47]: What is the purpose of this?

File: server/src/assets/game/script-game.js [hunk 2]
```diff
@@ -40,20 +44,35 @@ function disableAllDestinationButtons() {
...
 	}
 }
 
+function getImageSrcFlag(destinationSignal){
+	let FlagData = signalToFlagMapping[destinationSignal];
+	if(FlagData != null){
+		console.log(flagRootLocation + FlagData["file"]);
+		return "<img src='" + flagRootLocation + FlagData["file"] + "' style='width: 35px; height: 35px;'/>";
+	}
+	FlagData = signalToFlagMapping[destinationSignal.substring(0, destinationSignal.length-1)];
```
Comment (eyip002) [inline server/src/assets/game/script-game.js:57]: Why not use the proper names of the destination signals in signalToFlagMapping?

File: server/src/assets/game/script-game.js [hunk 3]
```diff
@@ -40,20 +44,35 @@ function disableAllDestinationButtons() {
...
+	if (FlagData != null){
+		return "<img src='" + flagRootLocation + FlagData["file"] + "' style='width: 35px; height: 35px;'/>";
+	}
+	return destinationSignal.substring(5, destinationSignal.length);
```
Comment (eyip002) [inline server/src/assets/game/script-game.js:61]: I don't understand why this is done

File: server/src/assets/game/script-game.js [hunk 4]
```diff
@@ -40,20 +44,35 @@ function disableAllDestinationButtons() {
...
+
+}
 function setDestinationButton(choice, route) {
 	// FIXME: Use signal-specific styles
 	const [destinationSignal, routeDetails] = unpackRoute(route);
-	
+
+	document.getElementById(`${destinationNamePrefix}${choice}`).innerHTML = getImageSrcFlag(destinationSignal);
```
Comment (eyip002) [inline server/src/assets/game/script-game.js:68]: For consistency, use the JQuery version: https://api.jquery.com/html/

**Review by eyip002 — CHANGES_REQUESTED** [review]

File: server/src/assets/game/script-game.js (hunk 1 above)
Comment (Noriakisana) [reply server/src/assets/game/script-game.js]: Removing the old svg - DEPRECATED

File: server/src/assets/game/script-game.js (hunk 2 above)
Comment (Noriakisana) [reply server/src/assets/game/script-game.js]: removed

File: server/src/assets/game/script-game.js (hunk 3 above)
Comment (Noriakisana) [reply server/src/assets/game/script-game.js]: rmeoved

**Comment by Noriakisana** [comment]
changed the dice view to completly css

- [ ] destination symbol generation

**Review by eyip002 — COMMENTED** [review]
The dynamic generation of the destination flags does work, but it is still too complicated. The location and sizes of the dots in the flag do not scale well with the screen size. The dots can become very small and the padding around the dice can become too large.

File: server/src/assets/game/SignalToFlag.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
+const signalToFlagFull = {
+    "signal4": {
+        "fillColor": "black",
+        "backgroundColor": "white",
+        "cssClassForColor": "bg-white-black",
+        "number": "1",
+        "numberClass": "one-face"
```
Comment (eyip002) [inline server/src/assets/game/SignalToFlag.js:7]: Should just be a single string that lists the visibility of each of the six dots, the circle colour, border colour, and background colour.

**Review by eyip002 — COMMENTED** [review]
The destination button should be the flag itself.

File: server/src/assets/game/driver-game.html [hunk 6]
```diff
@@ -13,13 +13,53 @@
...
+	
+	<symbol id="flagDef">
+		<rect width="100" height="100"></rect>
+		<circle class="topLeft" cx="25" cy="25" r="10"></circle>
+		<circle class="topRight" cx="75" cy="25" r="10"></circle>
+		<circle class="centre" cx="50" cy="50" r="10"></circle>
+		<circle class="bottomLeft" cx="25" cy="75" r="10"></circle>
+		<circle class="bottomRight" cx="75" cy="75" r="10"></circle>
+	</symbol>
+</svg>
```
Comment (eyip002) [inline server/src/assets/game/driver-game.html:61]: SVG template for the destination flag.

File: server/src/assets/game/flags-swtbahn-full.js [hunk 7]
```diff
@@ -0,0 +1,242 @@
+const signalToFlagFull = {
+    "signal4": "flagThemeBlack flagOutline flagOne",
+    "signal18": "flagThemeBlack flagOutline flagTwo",
```
Comment (eyip002) [inline server/src/assets/game/flags-swtbahn-full.js:3]: Examples of the mapping strings that should be generated instead.
Each string defines the colour theme, the type of flag design, and the flag symbol.

File: server/src/assets/game/script-game.js [hunk 8]
```diff
@@ -718,9 +772,15 @@ function initialise() {
 	});
 }
 
-$(document).ready(
-	() => initialise()
-);
+function debugDestinationFlags() {
+    $('#trainSelection').hide();
+	$('#destinationsForm').show();
+}
```
Comment (eyip002) [inline server/src/assets/game/script-game.js:778]: Function to help with debugging. Hides the train selection, and shows the destination buttons.
**Delete this after you have finished debugging!**

File: server/src/assets/game/script-game.js [hunk 9]
```diff
@@ -718,9 +772,15 @@ function initialise() {
...
+
+$(document).ready(() => {
+	initialise();
+	debugDestinationFlags();
```
Comment (eyip002) [inline server/src/assets/game/script-game.js:782]: **Delete this line after you have finished debugging!**

File: server/src/assets/game/style-game.css [hunk 10]
```diff
@@ -49,3 +49,106 @@ body {
...
+  --flag-fill: white;
+  --flag-circle-fill: var(--flag-theme);
+  --flag-stroke: var(--flag-theme);
+}
+
+.flagFilled {
+  --flag-fill: var(--flag-theme);
+  --flag-circle-fill: white;
+  --flag-stroke: var(--flag-theme);
+}
```
Comment (eyip002) [inline server/src/assets/game/style-game.css:103]: Colour theme and design types for the destination flag

File: server/src/assets/game/style-game.css [hunk 11]
```diff
@@ -49,3 +49,106 @@ body {
...
+  --flag-bottomRight-fill: var(--flag-circle-fill);
+}
+
+.flagFive {
+  --flag-topLeft-fill: var(--flag-circle-fill);
+  --flag-topRight-fill: var(--flag-circle-fill);
+  --flag-centre-fill: var(--flag-circle-fill);
+  --flag-bottomLeft-fill: var(--flag-circle-fill);
+  --flag-bottomRight-fill: var(--flag-circle-fill);
+}
```
Comment (eyip002) [inline server/src/assets/game/style-game.css:152]: Flag symbols

**Review by eyip002 — CHANGES_REQUESTED** [review]
With my latest push, I have added an SVG template for destination flag, added CSS styles for the SVG template, added examples of displaying the destination flags in HTML, and added examples of how the destination-flag mapping file should look like. 

Please update the implementation based on the examples.

File: server/src/assets/game/driver-game.html [hunk 12]
```diff
@@ -77,20 +117,36 @@ <h5 class="card-title">Regional ODEG</h5>
...
+				</svg>
+				<svg class="flag" viewBox="0 0 100 100">
+					<use id="destination1" href="#flagDef" class="flagThemeGreen flagOutline flagZero" value=""></use>
+				</svg>
+				<svg class="flag" viewBox="0 0 100 100">
+					<use id="destination2" href="#flagDef" class="flagThemeRed flagOutline flagThree" value=""></use>
+				</svg>
+				<svg class="flag" viewBox="0 0 100 100">
+					<use id="destination3" href="#flagDef" class="flagThemeBlue flagFilled flagFive" value=""></use>
+				</svg>
```
Comment (eyip002) [inline server/src/assets/game/driver-game.html:134]: Examples of using the SVG template to display different destination flags.

Comment (eyip002) [inline server/src/assets/game/driver-game.html:134]: Examples of using the SVG template to display different destination flags.

File: server/src/assets/game/driver-game.html [hunk 13]
```diff
@@ -77,20 +117,36 @@ <h5 class="card-title">Regional ODEG</h5>
...
+				<svg class="flag" viewBox="0 0 100 100">
+					<use id="destination5" href="#flagDef" class="flagThemeBlack flagFilled flagZero flagThemeDisabled" value=""></use>
+				</svg>
+				<svg class="flag" viewBox="0 0 100 100">
+					<use id="destination6" href="#flagDef" class="flagThemeRed flagOutline flagFive flagThemeDisabled" value=""></use>
+				</svg>
+				<svg class="flag" viewBox="0 0 100 100">
+					<use id="destination7" href="#flagDef" class="flagThemeBlank" value=""></use>
+				</svg>
```
Comment (eyip002) [inline server/src/assets/game/driver-game.html:149]: Examples of using the SVG template to display different destination flags.

**Review by eyip002 — APPROVED** [review]

**Pull request merged**